packages/plugin-react/src/fast-refresh.ts

```diff
@@ -58,13 +58,13 @@ if (import.meta.hot && !inWebWorker) {
   window.$RefreshReg$ = prevRefreshReg;
   window.$RefreshSig$ = prevRefreshSig;
 }`
-const sharedFooter = `
+const sharedFooter = (id: string) => `
 if (import.meta.hot && !inWebWorker) {
   RefreshRuntime.__hmr_import(import.meta.url).then((currentExports) => {
-    RefreshRuntime.registerExportsForReactRefresh(__SOURCE__, currentExports);
+    RefreshRuntime.registerExportsForReactRefresh("${id}", currentExports);
```

Comment:
```suggestion
    RefreshRuntime.registerExportsForReactRefresh(${JSON.stringify(id)}, currentExports);
```
I think this would be more safer as `id` can contain `"`/`\`.